framework/src/outputs/FileOutput.C

```diff
@@ -74,12 +77,24 @@ FileOutput::FileOutput(const InputParameters & parameters) :
     _file_base += buffer;
   }
 
-  // Check the file directory of file_base
+  // Check the file directory of file_base and create if needed
   std::string base = "./" + _file_base;
   base = base.substr(0, base.find_last_of('/'));
   if (access(base.c_str(), W_OK) == -1)
-    mooseError("Can not write to directory: " + base + " for file base: " + _file_base);
-
+  {
+    //Directory does not exist. Loop through incremental directories and create as needed.
+    size_t pos = 0;
+    std::string inc_path;
+    do
```

Comment:
>Or... boost:: filesystem (path iterators)... ;-)

You really DO like to antogonize don't you? :smile: 

There are already other functions in `MooseUtils` related to paths... this should be one too... but why can't whatever it is be based on `tokenize()`?  All it needs is a boolean parameter for honoring escape characters.

Seriously though... we're overdesigning this.  What the current code does is covered perfectly by the current `tokenize()`